### PR TITLE
Add Go 1.9beta2 to travis config

### PR DESCRIPTION
**Pull request by lucas-clemente**

### CODE REVIEW

**Comment by marten-seemann** [comment]
Oh, `gimme` finally has Go 1.9. Great!

Maybe this is a good moment to drop 1.7? What do you think?

File: .travis.yml [hunk 1]
```diff
@@ -12,6 +12,7 @@ services:
 go:
   - 1.7.x
   - 1.8.x
+  - 1.9beta2
```
Comment (marten-seemann) [inline .travis.yml]: Is there 1.9beta.x? Would save us a lot of commits until the final version.

**Comment by lucas-clemente** [comment]
I think gimme supports all versions that are on the download server, right?

**Comment by marten-seemann** [comment]
Not sure about that. I tried adding 1.9 a couple of days ago, and it didn't work. But maybe I got something else wrong, I didn't spend any time debugging this.

**Comment by lucas-clemente** [comment]
The failing test is caused by https://github.com/golang/go/commit/8f6d68ebaa660c6db8a87d418e95f8c0d3a221e4. I pushed a fix, @marten-seemann please review :)

File: h2quic/request_writer_test.go [hunk 2]
```diff
@@ -108,6 +108,10 @@ var _ = Describe("Request", func() {
 		req.AddCookie(cookie2)
 		rw.WriteRequest(req, 11, true, false)
 		_, headerFields := decode(headerStream.dataWritten.Bytes())
-		Expect(headerFields).To(HaveKeyWithValue("cookie", "Cookie #1=Value #1; Cookie #2=Value #2"))
+		// TODO(lclemente): Remove Or() once we drop support for Go 1.8.
+		Expect(headerFields).To(Or(
+			HaveKeyWithValue("cookie", "Cookie #1=Value #1; Cookie #2=Value #2"),
+			HaveKeyWithValue("cookie", `Cookie #1="Value #1"; Cookie #2="Value #2"`),
```
Comment (marten-seemann) [inline h2quic/request_writer_test.go]: Please add a comment here why we need this `Or`.

Comment (lucas-clemente) [inline h2quic/request_writer_test.go]: Isn't the todo + blame enough?

Comment (marten-seemann) [inline h2quic/request_writer_test.go]: Somehow I must have missed that line. Sorry!

**Comment by lucas-clemente** [comment]
CI ran through now, so I'm merging this.

**Pull request merged**